src/main/java/com/iota/iri/service/milestone/impl/LatestMilestoneTrackerImpl.java

```diff
@@ -115,41 +114,24 @@
     private boolean initialized = false;
 
     /**
-     * <p>
-     * This method initializes the instance and registers its dependencies.
-     * </p>
-     * <p>
-     * It simply stores the passed in values in their corresponding private properties and bootstraps the latest
-     * milestone with values for the latest milestone that can be found quickly.
-     * </p>
-     * <p>
-     * Note: Instead of handing over the dependencies in the constructor, we register them lazy. This allows us to have
-     *       circular dependencies because the instantiation is separated from the dependency injection. To reduce the
-     *       amount of code that is necessary to correctly instantiate this class, we return the instance itself which
-     *       allows us to still instantiate, initialize and assign in one line - see Example:
-     * </p>
-     *       {@code latestMilestoneTracker = new LatestMilestoneTrackerImpl().init(...);}
-     *
      * @param tangle Tangle object which acts as a database interface
      * @param snapshotProvider manager for the snapshots that allows us to retrieve the relevant snapshots of this node
      * @param milestoneService contains the important business logic when dealing with milestones
      * @param milestoneSolidifier manager that takes care of solidifying milestones
      * @param config configuration object which allows us to determine the important config parameters of the node
-     * @return the initialized instance itself to allow chaining
      */
-    public LatestMilestoneTrackerImpl init(Tangle tangle, SnapshotProvider snapshotProvider,
-            MilestoneService milestoneService, MilestoneSolidifier milestoneSolidifier, IotaConfig config) {
-
+    public LatestMilestoneTrackerImpl(Tangle tangle, SnapshotProvider snapshotProvider, MilestoneService milestoneService,
+                                      MilestoneSolidifier milestoneSolidifier, IotaConfig config) {
```

Comment:
Since we are dealing with properly injecting parameters to constructor, maybe it makes sense for you to change `IotaConfig` -> `MilestoneConfig`

Since this is not your fault, you can skip this request

Reply:
Yes, let's do this in another PR. I tried not to change anything that wasn't really necessary.